### DISCUSSION TITLE
`test-bot` workflow fails with `Error: Invalid usage: Did not find any formulae or commits to test!`

### DISCUSSION THREAD

**Discussion by Southpaw1496** [Homebrew/discussions#3918] [Tap maintenance and Homebrew development] [answered]
I followed [this article](https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/) to set up my own tap, and I found that `tap-new` helpfully creates a GitHub Actions workflow for compiling formulas, but when I opened a Pull Request to run it, it failed with the error "Error: Invalid usage: Did not find any formulae or commits to test!"

The Pull Request adds the following formula, which builds and runs successfully on my local machine using `brew install --build-from-source`. It doesn't have any real tests because the app doesn't have any CLI functions as far as I can tell:
```rb
# Documentation: https://docs.brew.sh/Formula-Cookbook
 #                https://rubydoc.brew.sh/Formula
 # PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
 class Quilt < Formula
   desc ""
   homepage ""
   url "https://github.com/QuiltMC/quilt-native-installer/archive/refs/tags/v0.1.1.tar.gz"
   sha256 "42d80580dc11e3a883b06633450f8121de704cb0ba24ac56d067f898473eddaa"
   license "Apache-2"

   depends_on "rust" => :build

   def install
     # ENV.deparallelize  # if your formula fails when building in parallel
     system "cargo", "install", *std_cargo_args
 	 mv "#{bin}/quilt-installer", "#{bin}/quilt"
   end

   test do
     # `test do` will create, run in and delete a temporary directory.
     #
     # This test will fail and we won't accept that! For Homebrew/homebrew-core
     # this will need to be a test that verifies the functionality of the
     # software. Run the test with `brew test quilt`. Options passed
     # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
     #
     # The installed folder is not in the path, so use the entire path to any
     # executables being tested: `system "#{bin}/program", "do", "something"`.
     system "true"
   end
 end
```
Here's the GitHub Actions log of the most recently failed run: https://gist.github.com/Southpaw1496/432d354320c00ae74d9afac858e87b80

**Comment by SMillerDev** [comment]
Well, your tap doesn't have the correct name for a tap, which requires `<owner>/homebrew-<tap name>` and the github action also seems to think nothing changed in the pull request.

**Reply by Southpaw1496** [reply]
I thought the tap naming was a recommendation, not a requirement.

I definitely have changed files in the pull request, why might it think I haven't?

https://github.com/Southpaw1496/homebrew/pull/1

**Comment by Southpaw1496** [comment, marked as answer]
I fixed the error by deleting the Pull Request and branch, and starting afresh.

Tips for those from the future to help you on your journey: 
- Make sure to run `brew audit --online --strict` and fixing the issues it highlights, even if you're not planning to upload to core. It catches many issues that might affect your formula's validity, and Homebrew's ability to find it.
- If you want to rename your formula halfway through, don't just rename the file, do it properly and create a new formula using `brew create`. You can use `--set-name` if you don't like the name that Homebrew sets based on the inputted TAR file.
